core/include/emulate.h

```diff
@@ -0,0 +1,44 @@
+#ifndef HAX_CORE_EMULATE_H_
+#define HAX_CORE_EMULATE_H_
+
+#include "../include/hax.h"
+#include "emulate_ops.h"
+
+/* Access completed successfully */
+#define EM_CONTINUE        0
+
+struct em_operand_t {
+    unsigned int bytes;
+    enum {
+        OP_NONE,
+        OP_REG,
+        OP_MEM,
+        OP_IMM,
+    } type;
+    struct segmented_addr_t {
+        HAX_VADDR_T addr;
```

Comment:
Is `HAX_VADDR_T` defined anywhere? Do you mean `vaddr_t`?

Reply:
https://github.com/intel/haxm/blob/428a9dae21742b4d8b3ccbacaeb11e6d37c542b5/include/windows/hax_types_windows.h#L139-L147

But probably `vaddr_t` is more convenient.

Reply:
Right! Sorry I missed the definition (never used it myself).